File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    private static final int PREPARATION_MINUTES_PER_LAYER = 2;
    private static final int EXPECTED_TIME_IN_OVEN = 40;
    
    public static int expectedMinutesInOven() {
        return EXPECTED_TIME_IN_OVEN;
    }

    public static int remainingMinutesInOven(int actualMinutesInOven) {
        return expectedMinutesInOven() - actualMinutesInOven;
    }

    public static int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * PREPARATION_MINUTES_PER_LAYER;
    }

    public static int totalTimeInMinutes(int numberOfLayers, int actualMinutesInOven) {
        return preparationTimeInMinutes(numberOfLayers) + actualMinutesInOven;
    }
}

File: solutions/elixir/lasagna/1/lib/lasagna.ex
defmodule Lasagna do

  def expected_minutes_in_oven, do: 40

  def remaining_minutes_in_oven(elapsed) do
    expected_minutes_in_oven() - elapsed
  end

  def preparation_time_in_minutes(layers) do
    prep_time_per_layer = 2
    layers * prep_time_per_layer
  end

  def total_time_in_minutes(layers, elapsed) do
    preparation_time_in_minutes(layers) + elapsed
  end

  def alarm, do: "Ding!"
end
